packages/shared/src/domain/contact/phone.ts

```diff
@@ -0,0 +1,51 @@
+import { stripNonNumericChars } from "../../common/string";
+
+export const phoneLength = 10;
+export const examplePhoneNumber = "1231231234";
+
+export function isPhoneValid(phone: string): boolean {
+  if (!phone) return false;
+  if (phone.length !== phoneLength) return false;
+  if (phone.match(/\D/)) return false; // needed as parseInt ignores non-numeric suffix
+  const phoneNumber = parseInt(phone, 10);
+  return !isNaN(phoneNumber);
+}
+
+/**
+ * Normalize a telephone number to a 10-digit string. Removes all non-numeric characters.
+ * Removes the first digit (country code) if the number is 11 digits long and starts with a 1.
+ * Returns the original number (without non-numeric chars) if:
+ * - the number is less than 10 digits long
+ * - the number is more than 11 digits long
+ * - the number is 11 digits long and does not start with a 1
+ * @param telephone the phone number to normalize
+ */
+export function normalizePhoneNumber(telephone: string): string;
+/**
+ * Normalize a telephone number to a 10-digit string. Removes all non-numeric characters.
+ * Removes the first digit (country code) if the number is 11 digits long and starts with a 1.
+ * Returns the original number (without non-numeric chars) if:
+ * - the number is less than 10 digits long
+ * - the number is more than 11 digits long and autofix is false
+ * - the number is 11 digits long, it does not start with a 1  and autofix is false
+ * @param telephone the phone number to normalize
+ * @param telephone the phone number to be normalized
+ * @param autofix whether to remove extra digits or not (default: false)
+ */
+export function normalizePhoneNumber(telephone: string, autofix: boolean): string;
+export function normalizePhoneNumber(telephone: string, autofix = false): string {
```

Comment:
I don't agree with this `autofix` param personally. It's only being used / not used in validation (technically also MPI but I don't know why it's not being used there). I'd expect a consistent output out of this that would pass our validation.

Reply:
The output is consistent based on the setup of the function.

If we don't use normalization on the route level, then I agree we don't need it and can always "autofix".

Reply:
My point is that the normalization function passes back things longer than 10 characters, but our validation wouldn't allow this... this doesn't make sense to me (see comment above about normalization).

Reply:
Removed autofix. 👍 

WRT to normalization and validation being compatible, these are separate functions, so I think its totally fine for normalization to return something that's not valid - otherwise we'd be coupling those functions. For example, we could have diff validation functions and a single normalization one.

Reply:
I think ultimately it depends on how far you take "normalization" -- it can either be: remove non-numeric characters or remove non-numeric characters and fill / slice to match our DB schema. Right now it's a bit of a middle ground. It slices longer than 10 digits, but doesn't fill on shorter than 10 digits (which is fine for now).